daphne/src/roles.rs

```diff
@@ -231,6 +237,24 @@ pub trait DapLeader<S>: DapAuthorizedSender<S> + DapAggregator<S> {
             collect_req.agg_param
         );
 
+        // Check that the batch interval is not too wide or too far into the past/future.
```

Comment:
Looks good. One more thing: Since the check is identical in the Leader and Helper, we should try to deduplicate it. There are two ways to do this:
* Create a macro and call it from both methods
* Add a derived method to `DapAggregator` that performs this check and call it from both methods.

Reply:
Good point. I've created a macro, since I didn't want to put too much methods in DapAggregator.